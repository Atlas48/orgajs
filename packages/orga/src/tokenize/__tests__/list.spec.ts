import tok from "./tok"

describe("tokenize list item", () => {
  it("knows list items", () => {
    // unordered
    expect(tok("- buy milk")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "buy milk",
          "type": "text.plain",
          "value": "buy milk",
        },
      ]
    `)
    expect(tok("+ buy milk")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "+",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "buy milk",
          "type": "text.plain",
          "value": "buy milk",
        },
      ]
    `)
    // ordered
    expect(tok("1. buy milk")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "1.",
          "indent": 0,
          "ordered": true,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "buy milk",
          "type": "text.plain",
          "value": "buy milk",
        },
      ]
    `)
    expect(tok("12. buy milk")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "12.",
          "indent": 0,
          "ordered": true,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "buy milk",
          "type": "text.plain",
          "value": "buy milk",
        },
      ]
    `)
    expect(tok("123) buy milk")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "123)",
          "indent": 0,
          "ordered": true,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "buy milk",
          "type": "text.plain",
          "value": "buy milk",
        },
      ]
    `)
    // checkbox
    expect(tok("- [x] buy milk checked")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "[x]",
          "checked": true,
          "type": "list.item.checkbox",
        },
        Object {
          "_text": "buy milk checked",
          "type": "text.plain",
          "value": "buy milk checked",
        },
      ]
    `)
    expect(tok("- [X] buy milk checked")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "[X]",
          "checked": true,
          "type": "list.item.checkbox",
        },
        Object {
          "_text": "buy milk checked",
          "type": "text.plain",
          "value": "buy milk checked",
        },
      ]
    `)
    expect(tok("- [-] buy milk checked")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "[-]",
          "checked": true,
          "type": "list.item.checkbox",
        },
        Object {
          "_text": "buy milk checked",
          "type": "text.plain",
          "value": "buy milk checked",
        },
      ]
    `)
    expect(tok("- [ ] buy milk unchecked")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "[ ]",
          "checked": false,
          "type": "list.item.checkbox",
        },
        Object {
          "_text": "buy milk unchecked",
          "type": "text.plain",
          "value": "buy milk unchecked",
        },
      ]
    `)
    // indent
    expect(tok("  - buy milk")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 2,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "buy milk",
          "type": "text.plain",
          "value": "buy milk",
        },
      ]
    `)
    // tag
    expect(tok("- item1 :: description here")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "item1",
          "type": "list.item.tag",
          "value": "item1",
        },
        Object {
          "_text": "description here",
          "type": "text.plain",
          "value": "description here",
        },
      ]
    `)
    expect(tok("- item2\n :: description here")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "item2",
          "type": "text.plain",
          "value": "item2",
        },
        Object {
          "_text": "
      ",
          "type": "newline",
        },
        Object {
          "_text": ":: description here",
          "type": "text.plain",
          "value": ":: description here",
        },
      ]
    `)
    expect(tok("- [x] item3 :: description here")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "[x]",
          "checked": true,
          "type": "list.item.checkbox",
        },
        Object {
          "_text": "item3",
          "type": "list.item.tag",
          "value": "item3",
        },
        Object {
          "_text": "description here",
          "type": "text.plain",
          "value": "description here",
        },
      ]
    `)
  })

  it("knows these are not list items", () => {
    expect(tok("-not item")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-not item",
          "type": "text.plain",
          "value": "-not item",
        },
      ]
    `)
    expect(tok("1.not item")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "1.not item",
          "type": "text.plain",
          "value": "1.not item",
        },
      ]
    `)
    expect(tok("8)not item")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "8)not item",
          "type": "text.plain",
          "value": "8)not item",
        },
      ]
    `)
    expect(tok("8a) not item")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "8a) not item",
          "type": "text.plain",
          "value": "8a) not item",
        },
      ]
    `)
  })
})
