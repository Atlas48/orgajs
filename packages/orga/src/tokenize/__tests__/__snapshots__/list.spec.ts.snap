// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize list item knows list items 1`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk",
  },
]
`;

exports[`tokenize list item knows list items 2`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk",
  },
]
`;

exports[`tokenize list item knows list items 3`] = `
Array [
  Object {
    "indent": 0,
    "ordered": true,
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 3,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk",
  },
]
`;

exports[`tokenize list item knows list items 4`] = `
Array [
  Object {
    "indent": 0,
    "ordered": true,
    "position": Object {
      "end": Object {
        "column": 3,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 4,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk",
  },
]
`;

exports[`tokenize list item knows list items 5`] = `
Array [
  Object {
    "indent": 0,
    "ordered": true,
    "position": Object {
      "end": Object {
        "column": 4,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 5,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk",
  },
]
`;

exports[`tokenize list item knows list items 6`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "checked": true,
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "list.item.checkbox",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 6,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk checked",
  },
]
`;

exports[`tokenize list item knows list items 7`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "checked": true,
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "list.item.checkbox",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 6,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk checked",
  },
]
`;

exports[`tokenize list item knows list items 8`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "checked": true,
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "list.item.checkbox",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 6,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk checked",
  },
]
`;

exports[`tokenize list item knows list items 9`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "checked": false,
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "list.item.checkbox",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 24,
        "line": 0,
      },
      "start": Object {
        "column": 6,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk unchecked",
  },
]
`;

exports[`tokenize list item knows list items 10`] = `
Array [
  Object {
    "indent": 2,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 3,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 4,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "buy milk",
  },
]
`;

exports[`tokenize list item knows list items 11`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "list.item.tag",
    "value": " :: ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 27,
        "line": 0,
      },
      "start": Object {
        "column": 11,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "description here",
  },
]
`;

exports[`tokenize list item knows list items 12`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "item2",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 0,
        "line": 1,
      },
      "start": Object {
        "column": 7,
        "line": 0,
      },
    },
    "type": "newline",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 1,
      },
      "start": Object {
        "column": 1,
        "line": 1,
      },
    },
    "type": "text.plain",
    "value": ":: description here",
  },
]
`;

exports[`tokenize list item knows list items 13`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "checked": true,
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "list.item.checkbox",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 6,
        "line": 0,
      },
    },
    "type": "list.item.tag",
    "value": " :: ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 31,
        "line": 0,
      },
      "start": Object {
        "column": 15,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "description here",
  },
]
`;

exports[`tokenize list item knows these are not list items 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 9,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "-not item",
  },
]
`;

exports[`tokenize list item knows these are not list items 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "1.not item",
  },
]
`;

exports[`tokenize list item knows these are not list items 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "8)not item",
  },
]
`;

exports[`tokenize list item knows these are not list items 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "8a) not item",
  },
]
`;
